packages/core/admin/admin/src/content-manager/services/documents.ts

```diff
@@ -282,61 +281,59 @@ const documentApi = contentManagerApi.injectEndpoints({
         method: 'POST',
         data: body,
       }),
-      invalidatesTags: (_res, _error, { model, ids }) =>
-        ids.map((id) => ({ type: 'Document', id: `${model}_${id}` })),
+      invalidatesTags: (_res, _error, { model, documentIds }) =>
+        documentIds.map((id) => ({ type: 'Document', id: `${model}_${id}` })),
     }),
     updateDocument: builder.mutation<
       Contracts.CollectionTypes.Update.Response,
-      {
-        collectionType: string;
-        model: string;
-        id?: string;
-        data: Contracts.CollectionTypes.Update.Request['body'];
-        params?: Contracts.CollectionTypes.Update.Request['query'];
-      }
+      Pick<Contracts.CollectionTypes.Update.Params, 'model'> &
+        Partial<Pick<Contracts.CollectionTypes.Update.Params, 'documentId'>> & {
+          collectionType: string;
+          data: Contracts.CollectionTypes.Update.Request['body'];
+          params?: Contracts.CollectionTypes.Update.Request['query'];
+        }
     >({
-      query: ({ collectionType, model, id, data, params }) => ({
-        url: `/content-manager/${collectionType}/${model}${id ? `/${id}` : ''}`,
+      query: ({ collectionType, model, documentId, data, params }) => ({
+        url: `/content-manager/${collectionType}/${model}${documentId ? `/${documentId}` : ''}`,
         method: 'PUT',
         data,
         config: {
           params,
         },
       }),
-      invalidatesTags: (_result, _error, { collectionType, model, id }) => {
+      invalidatesTags: (_result, _error, { collectionType, model, documentId }) => {
         return [
-          { type: 'Document', id: collectionType !== SINGLE_TYPES ? `${model}_${id}` : model },
+          {
+            type: 'Document',
+            id: collectionType !== SINGLE_TYPES ? `${model}_${documentId}` : model,
+          },
         ];
       },
     }),
     unpublishDocument: builder.mutation<
       Contracts.CollectionTypes.Unpublish.Response,
-      {
-        collectionType: string;
-        model: string;
-        /**
-         * You don't pass the ID if the document is a single-type
-         */
-        id?: string;
-        params?: Contracts.CollectionTypes.Unpublish.Request['query'];
-        data: Contracts.CollectionTypes.Unpublish.Request['body'];
-      }
+      Pick<Contracts.CollectionTypes.Unpublish.Params, 'model'> &
+        Partial<Pick<Contracts.CollectionTypes.Unpublish.Params, 'documentId'>> & {
```

Comment:
Don't thing those are clearly exposed by strapi types atm :/ But it feels like Model and DocumentId should be somewhat more generic and imported from somewhere else, but don't know if in this specific case you need this to follow the contract

Reply:
yeah, i think theres some room to be desired here, but at the moment i've just tried to ensure we have one source of truth for the types, i had re-written bits when in reality i should have mutated existing things 🙃

Reply:
makes sense!